src/ai/defaultai_warfare.cc

```diff
@@ -981,32 +981,71 @@ bool DefaultAI::check_militarysites(const Time& gametime) {
-		          enemy_military_capacity) {
-			should_be_dismantled = true;
-		}
-	} else {
-		const uint16_t size_bonus =
-		   total_capacity * std::abs(management_data.get_military_number_at(89)) / 5;
-		if (bf.military_score_ + size_bonus < management_data.get_military_number_at(88) * 5 &&
-		    bf.area_military_capacity > static_cast<int16_t>(total_capacity)) {
-			should_be_dismantled = true;
-		}
+	// inputs for decisions
+	// bf.enemy_owned_land_nearby - decreases dismantle probability
+	// bf.unowned_land_nearby - decreases dismantle probability
+	// portspace_nearby - decreases dismantle probability
+
+	// bf.future_area_military_capacity - increases dismantle probability
+	// bf.military_loneliness - increases dismantle probability
+	// bf.future_military_loneliness - increases dismantle probability
+	// soldier_status_
+	// bf.military_score_ - decreases dismantle probability
+	// bf.own_military_presence - increases dismantle probability
+
+	// bf.enemy_military_presence - decreases dismantle probability
+	// bf.enemy_military_sites - - decreases dismantle probability
+
+	// We calculate the score - if positive, site will be dismounted
+	// We starting with positive number, most modificators are decreasing th score
+	int32_t dismantle_score = 3 * management_data.get_military_number_at(91) / 2 + 100;
+	dismantle_score -= 2 * management_data.neuron_pool[13].get_result_safe(
+	                          bf.enemy_owned_land_nearby / 2, kAbsValue);
+	dismantle_score -= (bf.enemy_owned_land_nearby > 0) ?
+                         std::abs(management_data.get_military_number_at(77)) :
+                         -std::abs(management_data.get_military_number_at(77));
+	dismantle_score -=
+	   management_data.neuron_pool[14].get_result_safe(bf.unowned_land_nearby / 2, kAbsValue);
+	dismantle_score -= (bf.unowned_land_nearby > 0) ?
+                         std::abs(management_data.get_military_number_at(99)) :
+                         -std::abs(management_data.get_military_number_at(99));
+	dismantle_score -= (bf.portspace_nearby == ExtendedBool::kTrue) ?
+                         std::abs(management_data.get_military_number_at(91)) :
+                         0;
+
+	dismantle_score +=
+	   management_data.neuron_pool[28].get_result_safe(bf.area_military_capacity, kAbsValue);
+	dismantle_score +=
+	   management_data.neuron_pool[57].get_result_safe(bf.future_area_military_capacity, kAbsValue);
+	dismantle_score -=
+	   management_data.neuron_pool[29].get_result_safe(bf.military_loneliness / 50, kAbsValue);
+	// dismantle_score -= management_data.neuron_pool[30].get_result_safe( // NOCOM uncomment this
+	//	                        bf.future_military_loneliness, kAbsValue);
+	if (soldier_status_ == SoldiersStatus::kBadShortage) {
+		dismantle_score += 2 * std::abs(management_data.get_military_number_at(77));
+	} else if (soldier_status_ == SoldiersStatus::kShortage) {
+		dismantle_score += std::abs(management_data.get_military_number_at(77));
```

Comment:
I meant to use 24 as magic number for bad shortage and another number (e.g.) for shortage to let the genetics decide on the real relation between both values rather then fix it  to 2. just to prevent fixed numbers and make it as succeptible as possible to genetics

Reply:
This means that bad shortage can get smaller malus then normal shortage - while genetics can cope with this and no tragedy it would be, I just think we should not even allow such possibility

Reply:
we still could keep a factor of 2 in front of the kBadShortage number. However it should work with using 1 number as well